src/Facade/ResourceBuilders/CurrencyResourceBuilder.cs

```diff
@@ -4,17 +4,28 @@
     using System.Collections.Generic;
 
     using Linn.Common.Facade;
+    using Linn.Purchasing.Domain.LinnApps.ExternalServices;
     using Linn.Purchasing.Domain.LinnApps.PurchaseOrders;
     using Linn.Purchasing.Resources;
 
     public class CurrencyResourceBuilder : IBuilder<Currency>
     {
+        private readonly ICurrencyPack currencyPack;
+
+        public CurrencyResourceBuilder(ICurrencyPack currencyPack)
+        {
+            this.currencyPack = currencyPack;
+        }
+
         public CurrencyResource Build(Currency entity, IEnumerable<string> claims)
         {
+            var exchangeRate = this.currencyPack.GetExchangeRate(entity.Code, "GBP");
```

Comment:
ooh hadn't thought of this in the resource builder, saves a lot of messing about